internal/handlers/create_short_url.go

```diff
@@ -23,17 +24,22 @@ func (h *Handler) CreateShortURL(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Создаем или получаем существующий короткий ID
-	var id string
-	var exists bool
-
-	// Сначала проверяем, есть ли такой URL в хранилище
-	id, exists = h.storage.FindByOriginalURL(originalURL)
-	if !exists {
-		// Если URL не найден, генерируем новый ID
-		id = h.shortener.Shorten(originalURL)
-		// Добавляем URL в хранилище
-		id, _ = h.storage.Add(id, originalURL)
+	// Генерируем новый ID и пытаемся добавить URL
+	id := h.shortener.Shorten(originalURL)
+	id, exists, err := h.storage.Add(id, originalURL)
+
+	if err != nil {
+		if err == database.ErrURLConflict {
+			// Если URL уже существует, возвращаем короткий URL с кодом конфликта
+			shortURL := h.shortener.BuildShortURL(id)
+			w.Header().Set("Content-Type", "text/plain")
+			w.WriteHeader(http.StatusConflict)
+			w.Write([]byte(shortURL))
+			return
+		}
+		logger.Logger.Error("Ошибка добавления URL", zap.Error(err))
+		http.Error(w, "Ошибка сохранения URL", http.StatusInternalServerError)
+		return
```

Comment:
Здесь так же можно использовать `errors.Is(...)` для проверки.
```suggestion
	if err != nil {
		if errors.Is(err, database.ErrURLConflict) {
			// Если URL уже существует, возвращаем короткий URL с кодом конфликта
			shortURL := h.shortener.BuildShortURL(id)
			w.Header().Set("Content-Type", "text/plain")
			w.WriteHeader(http.StatusConflict)
			w.Write([]byte(shortURL))
			return
		}
		logger.Logger.Error("Ошибка добавления URL", zap.Error(err))
		http.Error(w, "Ошибка сохранения URL", http.StatusInternalServerError)
		return
	}
```